src/clj/rems/api/blacklist.clj

```diff
@@ -26,6 +27,11 @@
          :blacklist/added-by schema-base/UserWithAttributes
          :blacklist/added-at DateTime))
 
+(defn user-or-resource-not-found-error [command]
+  (when-not (and (users/user-exists? (get-in command [:blacklist/user :userid]))
+                 (resource/ext-id-exists? (get-in command [:blacklist/resource :resource/ext-id])))
+    (not-found-json-response)))
```

Comment:
Is it not found or invalid? The resource in question is blacklist, and a parameter in the "command" is invalid. That would suggest 400 to me. If the URL is invalid, i.e. main resource missing, then 404?

Reply:
mayhaps HTTP 422? looking at this SO thread https://stackoverflow.com/questions/42143115/which-status-code-is-correct-404-or-400-and-when-to-use-either-of-these